app/models/symphony_curr_loc_request.rb

```diff
@@ -0,0 +1,53 @@
+require 'json'
+
+# API for retrieving the current location from a barcode via Symphony Web Services
+class SymphonyCurrLocRequest
+  include ActiveModel::Model  # allows initialization with a hash of attributes
+
+  attr_accessor :barcode
+
+  def current_location
+    return '' if json.blank?
+    json['fields']['currentLocation']['key'] || ''
```

Comment:
i was following a pattern here.  Not sure if we should silently return ''  if there was a problem.  (we log it, but ...)